IPs/RSKIP241.md

```diff
@@ -0,0 +1,120 @@
+
+### Mixing Dirty UTXOs
+
+The feeBumpWithInput() could be used to mix "dirty" UTXOs (for example, considered to come from illicit activity) into peg-out transactions. However, this feature cannot be reliably used to launder bitcoins, as the input added is fully paid to miners, and its value is supposed to be vey low. However, the Powpeg doesn't have (and since peg-ins are decentralized, it cannot have) any mean of avoiding the peg-ins with of illicit funds. Therefore feeBumpWithInput() cannot have a negative effect on the platform.
+
+### Input Selection Algorithm
+
+The input selection algorithm must do a reasonable effort to select inputs that allow 4X fee bumping. By reasonable we mean that the algorithm should still be linear, or the number of powpeg UTXOs, bounded. The algorithm should not add more inputs in order to satisfy this condition, as the cost in virtual bytes of a non-segwit input (~137 bytes) is higher than the cost of CPFP output (30 bytes). Adding more inputs only makes sense if they are all low valued, and they are using the peg-out transaction for input consolidation.
+
+## Backwards Compatibility
+
+This change is a hard-fork and therefore all full nodes must be updated. SPV light-clients and Block explorers do not need to be updated. 
+
+## Test Cases
+
+TBD
+
+## Security Considerations
+
+* The method used to create the new transaction forces it to be always a double-spend of the prior one, therefore this method cannot be used to perform two peg-outs at the price of one.
+
+* This proposal avoids merging new Bridge inputs when creating a fee bumping transaction. The reason is that the bridge does not know which one of the transactions (*T* or *T'*) will be finally included in the Bitcoin blockchain, and therefore all inputs must be marked as consumed. To unmark the unused inputs the Bridge would need to receive notification of the transaction *T* or *T'* when it is included in the Bitcoin blockchain, and this functionality is not part of the Bridge. But more importantly, an attacker may be able to force the bridge to create hundreds of variations  of the fee bumping transaction, locking all inputs and rendering the Bridge useless for many blocks.
+* FeeBumpWithInput cannot consume the input partially for fees because the Bridge doesn't know the input amount of the input given. To be sure of the input amount, the transaction would need to be segwit, because in segwit transactions the input amount is signed.
+* Is could possible for an attacker to block a a honest user from consuming a CPFPxxx output in a peg-out transaction by broadcasting a transaction that also consumes the CPFPxxx output, but for other reasons it is not mined. The rules that govern CPFP in the Bitcoin node must be carefully analyzed to disregard this attack.
```

Comment:
```suggestion
* It could possible for an attacker to block a a honest user from consuming a CPFPxxx output in a peg-out transaction by broadcasting a transaction that also consumes the CPFPxxx output, but for other reasons it is not mined. The rules that govern CPFP in the Bitcoin node must be carefully analyzed to disregard this attack.
```